sui/gmp.js

```diff
@@ -0,0 +1,201 @@
+const { saveConfig, printInfo } = require('../evm/utils');
+const { Command } = require('commander');
+const { Transaction } = require('@mysten/sui/transactions');
+const { bcs } = require('@mysten/sui/bcs');
+const { loadSuiConfig, getBcsBytesByObjectId } = require('./utils');
+const { ethers } = require('hardhat');
+const {
+    utils: { arrayify },
+} = ethers;
+
+const { addBaseOptions, addOptionsToCommands } = require('./cli-utils');
+const { getUnitAmount } = require('./amount-utils.js');
+const { getWallet, printWalletInfo, broadcast } = require('./sign-utils');
+const { discoveryStruct } = require('./types-utils.js');
+
+async function sendCommand(keypair, client, contracts, args, options) {
+    const [destinationChain, destinationAddress, feeAmount, payload] = args;
+    const params = options.params;
+
+    const [testConfig, gasServiceConfig] = contracts;
+    const gasServicePackageId = gasServiceConfig.address;
+    const singletonObjectId = testConfig.objects.singleton;
+    const channelId = testConfig.objects.channelId;
+
+    const unitAmount = getUnitAmount(feeAmount);
+    const walletAddress = keypair.toSuiAddress();
+    const refundAddress = options.refundAddress || walletAddress;
+
+    const tx = new Transaction();
+    const [coin] = tx.splitCoins(tx.gas, [unitAmount]);
+
+    tx.moveCall({
+        target: `${testConfig.address}::test::send_call`,
+        arguments: [
+            tx.object(singletonObjectId),
+            tx.pure(bcs.string().serialize(destinationChain).toBytes()),
+            tx.pure(bcs.string().serialize(destinationAddress).toBytes()),
+            tx.pure(bcs.vector(bcs.u8()).serialize(arrayify(payload)).toBytes()),
+        ],
+    });
+
+    if (gasServiceConfig) {
```

Comment:
can you change the `test` contract so that it pays the gas service during the send? Mimics our EVM gmp examples better. Also, let's rename the contract to `gmp` and mention that it's an example in it's readme.

Reply:
could do in a separate PR

Reply:
Make sense. I'll submit a pr to modify the test contract in the `axelar-cgp-sui` repo first, before making a follow-up update here. I've created 2 tickets to track them here:

- Modify the test contract in `axelar-cgp-sui` repo: https://axelarnetwork.atlassian.net/browse/AXE-4529
- Making a follow-up update here: https://axelarnetwork.atlassian.net/browse/AXE-4530